### DISCUSSION TITLE
Unable to increase pm.max_children for local Symfony web server

### DISCUSSION THREAD

**Discussion by grubersjoe** [symfony/symfony#45674] [Q&A] [answered]
Hey,

I'm running Symfony 5.4 and use Symfony's built in web server for local development. I start the server like this:
```sh
symfony server:start --no-tls
```

Sometimes I see this error message and consequently I want to raise this limit:

```
[PHP-FPM] WARNING| FPM    [pool www] server reached pm.max_children setting (5), consider raising it 
```

This is my local environment (Linux):

```sh
$ symfony local:php:list
+---------+-----------+---------+-------------+---------+---------+---------+
| Version | Directory |   PHP   |     PHP     |   PHP   | Server  | System? |
|         |           |   CLI   |     FPM     |   CGI   |         |         |
+---------+-----------+---------+-------------+---------+---------+---------+
| 8.1.3   | /usr      | bin/php | bin/php-fpm |         | PHP FPM | *       |
+---------+-----------+---------+-------------+---------+---------+---------+
```

So in order to increase this limit I have edited my `/etc/php/php-fpm.d/www.conf`:

```sh
grep "^[^#;]" /etc/php/php-fpm.d/www.conf
[www]
user = http
group = http
listen = /run/php-fpm/php-fpm.sock
listen.owner = http
listen.group = http
pm = dynamic
pm.max_children = 16
pm.start_servers = 8
pm.min_spare_servers = 2
pm.max_spare_servers = 8
pm.max_requests = 500
pm.status_path = /status
```

As you can see `pm.max_children` _is_ way higher, however, this configuration seems to be ignored completely. I still get the same error. Also I've tried setting this value in a local `www.conf` file, but again this is ignored.

**So my question is: how do I configure `php-fpm` when using the local Symfony server?**

Unfortunately, the Symfony documentation is not really helpful here.
https://symfony.com/doc/current/setup/symfony_server.html#enabling-php-fpm

**Comment by grubersjoe** [comment, marked as answer]
I've discovered that there is a config file in `~/.symfony5/php/ee9a79fa66ab93b45784b8b3a22ecefd9871ce5a/fpm-8.1.3.ini` that has the following contents:

```
[www]
listen = 127.0.0.1:33729
listen.allowed_clients = 127.0.0.1
pm = dynamic
pm.max_children = 5 # <<<
pm.start_servers = 2
pm.min_spare_servers = 1
pm.max_spare_servers = 3
pm.status_path = /__php-fpm-status__
```

So I suppose this is where the value of 5 stems, but I'm interested into setting some other value here.

Edit: Okay found the source for this, it seems this is hardcoded :neutral_face: .
https://github.com/symfony-cli/symfony-cli/blob/48ca923fa3adc2e725788626f02501dff52ddc6a/local/php/fpm.go#L95

Edit 2: Found this issue. I suppose I've been looking at the wrong place.
https://github.com/symfony-cli/symfony-cli/issues/96

**Comment by Herz3h** [comment]
I keep running into this issue as of late symfony cli versions...I have to restart it every time, 5 max requests seems so low. Would be nice to have a way to override it.

**Edit**: Simply cloned this repository, changed the hard coded max_children to a higher value and recompiled it using `go build`

**Reply by feldenla** [reply]
Please can you show how? I've been struggling with this issue since I updated my symfony command line.

**Comment by feldenla** [comment]
Why don't symfony CLI just pick the php-fpm configuration from your settings in /etc/php/8.4/fpm/pool.d/www.conf?
This would have saved a lot of time
